lib/catalog/copy_name_helper.rb

```diff
@@ -0,0 +1,34 @@
+module Catalog
+  module CopyNameHelper
```

Comment:
Small nit, since Rails already has a concept of a `helper` without the need to include a mixin, I would suggest something like `Catalog::NameAdjust` or `Catalog::NameCheck`.